{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# Packages and dependencies\n",
    "from ipywidgets import *\n",
    "import pandas as pd\n",
    "import glob\n",
    "import numpy as np\n",
    "import os\n",
    "import re\n",
    "import json\n",
    "import requests\n",
    "import csv\n",
    "import requests\n",
    "from io import StringIO \n",
    "import gzip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>TF1</th>\n",
       "      <th>cell1</th>\n",
       "      <th>gene1</th>\n",
       "      <th>organism</th>\n",
       "      <th>Enc16</th>\n",
       "      <th>CreedsDN</th>\n",
       "      <th>Overlap16down</th>\n",
       "      <th>TF2</th>\n",
       "      <th>cell2</th>\n",
       "      <th>...</th>\n",
       "      <th>CreedsDN.1</th>\n",
       "      <th>overlap15down</th>\n",
       "      <th>pvalue15down</th>\n",
       "      <th>TF4</th>\n",
       "      <th>cell4</th>\n",
       "      <th>gene4</th>\n",
       "      <th>CreedsUP</th>\n",
       "      <th>Overlap15up</th>\n",
       "      <th>pvalue15up</th>\n",
       "      <th>totalEncode15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SETDB1_K562_31</td>\n",
       "      <td>SETDB1</td>\n",
       "      <td>K562</td>\n",
       "      <td>31</td>\n",
       "      <td>hg19</td>\n",
       "      <td>956</td>\n",
       "      <td>345</td>\n",
       "      <td>2</td>\n",
       "      <td>SETDB1</td>\n",
       "      <td>K562</td>\n",
       "      <td>...</td>\n",
       "      <td>345.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SETDB1</td>\n",
       "      <td>K562</td>\n",
       "      <td>31</td>\n",
       "      <td>252</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MYC_HepG2_1820</td>\n",
       "      <td>MYC</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>1820</td>\n",
       "      <td>hg19</td>\n",
       "      <td>2000</td>\n",
       "      <td>217</td>\n",
       "      <td>2</td>\n",
       "      <td>MYC</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>...</td>\n",
       "      <td>217.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MYC</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>1820</td>\n",
       "      <td>353</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CTCF_astrocyteofthecerebellum_124</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>astrocyteofthecerebellum</td>\n",
       "      <td>124</td>\n",
       "      <td>hg19</td>\n",
       "      <td>2000</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>astrocyteofthecerebellum</td>\n",
       "      <td>...</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>astrocyteofthecerebellum</td>\n",
       "      <td>124</td>\n",
       "      <td>314</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CTCF_bronchialepithelialcell_124</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>bronchialepithelialcell</td>\n",
       "      <td>124</td>\n",
       "      <td>hg19</td>\n",
       "      <td>2000</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>bronchialepithelialcell</td>\n",
       "      <td>...</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>bronchialepithelialcell</td>\n",
       "      <td>124</td>\n",
       "      <td>314</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CTCF_cardiacfibroblast_124</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>cardiacfibroblast</td>\n",
       "      <td>124</td>\n",
       "      <td>hg19</td>\n",
       "      <td>2000</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>cardiacfibroblast</td>\n",
       "      <td>...</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>cardiacfibroblast</td>\n",
       "      <td>124</td>\n",
       "      <td>314</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         description     TF1                     cell1  gene1  \\\n",
       "0                     SETDB1_K562_31  SETDB1                      K562     31   \n",
       "1                     MYC_HepG2_1820     MYC                     HepG2   1820   \n",
       "2  CTCF_astrocyteofthecerebellum_124    CTCF  astrocyteofthecerebellum    124   \n",
       "3   CTCF_bronchialepithelialcell_124    CTCF   bronchialepithelialcell    124   \n",
       "4         CTCF_cardiacfibroblast_124    CTCF         cardiacfibroblast    124   \n",
       "\n",
       "  organism  Enc16  CreedsDN  Overlap16down     TF2                     cell2  \\\n",
       "0     hg19    956       345              2  SETDB1                      K562   \n",
       "1     hg19   2000       217              2     MYC                     HepG2   \n",
       "2     hg19   2000       286              1    CTCF  astrocyteofthecerebellum   \n",
       "3     hg19   2000       286              1    CTCF   bronchialepithelialcell   \n",
       "4     hg19   2000       286              1    CTCF         cardiacfibroblast   \n",
       "\n",
       "       ...        CreedsDN.1 overlap15down  pvalue15down     TF4  \\\n",
       "0      ...             345.0             0           1.0  SETDB1   \n",
       "1      ...             217.0             0           1.0     MYC   \n",
       "2      ...             286.0             0           1.0    CTCF   \n",
       "3      ...             286.0             0           1.0    CTCF   \n",
       "4      ...             286.0             0           1.0    CTCF   \n",
       "\n",
       "                      cell4 gene4  CreedsUP  Overlap15up  pvalue15up  \\\n",
       "0                      K562    31       252            0         1.0   \n",
       "1                     HepG2  1820       353            0         1.0   \n",
       "2  astrocyteofthecerebellum   124       314            0         1.0   \n",
       "3   bronchialepithelialcell   124       314            0         1.0   \n",
       "4         cardiacfibroblast   124       314            0         1.0   \n",
       "\n",
       "   totalEncode15  \n",
       "0              0  \n",
       "1              0  \n",
       "2              0  \n",
       "3              0  \n",
       "4              0  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Nao adicionar\n",
    "\n",
    "# Calculate Fisher Exact Test\n",
    "\n",
    "# Criate python dict, use it to generate and run the queries\n",
    "\n",
    "encpy = pd.read_csv(\"/Users/Carol/Desktop/tfA/dataTables/encode/encodeMasterRace.csv\")\n",
    "\n",
    "encpy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# for line in encpy:\n",
    "#      print(encpy[\"description\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (61,62,65,66,67,68,69,70,74,75,77,78,79,80,81,82,163,198,199,202,205,209,213,214,217,220,238,239,242,279,280,283,284,285,286,287,288,291,299,302,303,305,306,307,310,311,313,314,315,318,319,320,322,323,332,336,337,339,340,341,342,343,344,362,393,537,625,719,847,850,851,855,856,858,863,867,873,890,893,896,899,901,902,903,904,907,908,911,913,917,918,927,931) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REST_H1hESC_hg19</th>\n",
       "      <th>TAF1_HeLaS3_hg19</th>\n",
       "      <th>REST_HepG2_hg19</th>\n",
       "      <th>ZBTB7A_K562_hg19</th>\n",
       "      <th>SIX5_K562_hg19</th>\n",
       "      <th>SIN3A_SKNSH_hg19</th>\n",
       "      <th>CBX3_K562_hg19</th>\n",
       "      <th>TRIM28_K562_hg19</th>\n",
       "      <th>JUND_SKNSH_hg19</th>\n",
       "      <th>JUND_T47D_hg19</th>\n",
       "      <th>...</th>\n",
       "      <th>TCF12_MELcellline_mm9</th>\n",
       "      <th>CHD2_ESE14_mm9</th>\n",
       "      <th>MAFK_ESE14_mm9</th>\n",
       "      <th>USF1_MELcellline_mm9</th>\n",
       "      <th>IRF4_CH12.LX_mm9</th>\n",
       "      <th>MEF2A_CH12.LX_mm9</th>\n",
       "      <th>MEF2A_MELcellline_mm9</th>\n",
       "      <th>TCF12_CH12.LX_mm9</th>\n",
       "      <th>USF1_CH12.LX_mm9</th>\n",
       "      <th>NRF1_CH12.LX_mm9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINC01420</td>\n",
       "      <td>POLK</td>\n",
       "      <td>MMP24</td>\n",
       "      <td>PTGES3L</td>\n",
       "      <td>CIAO1</td>\n",
       "      <td>SSTR5-AS1</td>\n",
       "      <td>ARFRP1</td>\n",
       "      <td>LOC221946</td>\n",
       "      <td>PSG8</td>\n",
       "      <td>ZSWIM8</td>\n",
       "      <td>...</td>\n",
       "      <td>Zfp607</td>\n",
       "      <td>Zfp526</td>\n",
       "      <td>Gm4285</td>\n",
       "      <td>Saraf</td>\n",
       "      <td>Slc39a9</td>\n",
       "      <td>Gm12191</td>\n",
       "      <td>Exosc4</td>\n",
       "      <td>Znhit1</td>\n",
       "      <td>Dtx3</td>\n",
       "      <td>Dolk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POLR2K</td>\n",
       "      <td>MMEL1</td>\n",
       "      <td>ZNF586</td>\n",
       "      <td>LOC100996351</td>\n",
       "      <td>MRPL10</td>\n",
       "      <td>PRKCSH</td>\n",
       "      <td>ABHD16A</td>\n",
       "      <td>DUS2</td>\n",
       "      <td>CLIP3</td>\n",
       "      <td>GBA</td>\n",
       "      <td>...</td>\n",
       "      <td>C530005A16Rik</td>\n",
       "      <td>Gm1976</td>\n",
       "      <td>Tram1</td>\n",
       "      <td>Dph5</td>\n",
       "      <td>Rrnad1</td>\n",
       "      <td>Sphk2</td>\n",
       "      <td>Nsrp1</td>\n",
       "      <td>Zscan22</td>\n",
       "      <td>Magt1</td>\n",
       "      <td>Fbln1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF528</td>\n",
       "      <td>PPP2R3C</td>\n",
       "      <td>THAP10</td>\n",
       "      <td>PNMA5</td>\n",
       "      <td>MRPL51</td>\n",
       "      <td>PPP1R7</td>\n",
       "      <td>MIR6718</td>\n",
       "      <td>NRXN2</td>\n",
       "      <td>FTSJ3</td>\n",
       "      <td>SNORD18B</td>\n",
       "      <td>...</td>\n",
       "      <td>Cox11</td>\n",
       "      <td>Mrps31</td>\n",
       "      <td>Erlin2</td>\n",
       "      <td>Polb</td>\n",
       "      <td>Abcb7</td>\n",
       "      <td>Znhit1</td>\n",
       "      <td>Mrpl1</td>\n",
       "      <td>Fam132b</td>\n",
       "      <td>Hdac5</td>\n",
       "      <td>8-Mar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATP6V0B</td>\n",
       "      <td>LOC101927156</td>\n",
       "      <td>LINC00893</td>\n",
       "      <td>ESRRA</td>\n",
       "      <td>MAGEA8-AS1</td>\n",
       "      <td>MIR6866</td>\n",
       "      <td>TBCCD1</td>\n",
       "      <td>MGC45922</td>\n",
       "      <td>SNORD78</td>\n",
       "      <td>DUOXA1</td>\n",
       "      <td>...</td>\n",
       "      <td>C87436</td>\n",
       "      <td>Eaf1</td>\n",
       "      <td>3110082I17Rik</td>\n",
       "      <td>Nagpa</td>\n",
       "      <td>Uba7</td>\n",
       "      <td>Hnrnpr</td>\n",
       "      <td>Aaas</td>\n",
       "      <td>Rbm8a</td>\n",
       "      <td>Kidins220</td>\n",
       "      <td>Rasd1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARBP2</td>\n",
       "      <td>PSMA1</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>LINGO4</td>\n",
       "      <td>STK36</td>\n",
       "      <td>ST3GAL6-AS1</td>\n",
       "      <td>VPS9D1</td>\n",
       "      <td>NLRP4</td>\n",
       "      <td>C9orf43</td>\n",
       "      <td>LOC104054148</td>\n",
       "      <td>...</td>\n",
       "      <td>Cog4</td>\n",
       "      <td>Dnajc27</td>\n",
       "      <td>Mrpl1</td>\n",
       "      <td>Dph1</td>\n",
       "      <td>Tcea1</td>\n",
       "      <td>Zfp369</td>\n",
       "      <td>Fcgr1</td>\n",
       "      <td>Lrrc58</td>\n",
       "      <td>Vapa</td>\n",
       "      <td>Sgpp1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 935 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  REST_H1hESC_hg19 TAF1_HeLaS3_hg19 REST_HepG2_hg19 ZBTB7A_K562_hg19  \\\n",
       "0        LINC01420             POLK           MMP24          PTGES3L   \n",
       "1           POLR2K            MMEL1          ZNF586     LOC100996351   \n",
       "2           ZNF528          PPP2R3C          THAP10            PNMA5   \n",
       "3          ATP6V0B     LOC101927156       LINC00893            ESRRA   \n",
       "4           TARBP2            PSMA1          WRAP53           LINGO4   \n",
       "\n",
       "  SIX5_K562_hg19 SIN3A_SKNSH_hg19 CBX3_K562_hg19 TRIM28_K562_hg19  \\\n",
       "0          CIAO1        SSTR5-AS1         ARFRP1        LOC221946   \n",
       "1         MRPL10           PRKCSH        ABHD16A             DUS2   \n",
       "2         MRPL51           PPP1R7        MIR6718            NRXN2   \n",
       "3     MAGEA8-AS1          MIR6866         TBCCD1         MGC45922   \n",
       "4          STK36      ST3GAL6-AS1         VPS9D1            NLRP4   \n",
       "\n",
       "  JUND_SKNSH_hg19 JUND_T47D_hg19       ...        TCF12_MELcellline_mm9  \\\n",
       "0            PSG8         ZSWIM8       ...                       Zfp607   \n",
       "1           CLIP3            GBA       ...                C530005A16Rik   \n",
       "2           FTSJ3       SNORD18B       ...                        Cox11   \n",
       "3         SNORD78         DUOXA1       ...                       C87436   \n",
       "4         C9orf43   LOC104054148       ...                         Cog4   \n",
       "\n",
       "  CHD2_ESE14_mm9 MAFK_ESE14_mm9 USF1_MELcellline_mm9 IRF4_CH12.LX_mm9  \\\n",
       "0         Zfp526         Gm4285                Saraf          Slc39a9   \n",
       "1         Gm1976          Tram1                 Dph5           Rrnad1   \n",
       "2         Mrps31         Erlin2                 Polb            Abcb7   \n",
       "3           Eaf1  3110082I17Rik                Nagpa             Uba7   \n",
       "4        Dnajc27          Mrpl1                 Dph1            Tcea1   \n",
       "\n",
       "  MEF2A_CH12.LX_mm9 MEF2A_MELcellline_mm9 TCF12_CH12.LX_mm9 USF1_CH12.LX_mm9  \\\n",
       "0           Gm12191                Exosc4            Znhit1             Dtx3   \n",
       "1             Sphk2                 Nsrp1           Zscan22            Magt1   \n",
       "2            Znhit1                 Mrpl1           Fam132b            Hdac5   \n",
       "3            Hnrnpr                  Aaas             Rbm8a        Kidins220   \n",
       "4            Zfp369                 Fcgr1            Lrrc58             Vapa   \n",
       "\n",
       "  NRF1_CH12.LX_mm9  \n",
       "0             Dolk  \n",
       "1            Fbln1  \n",
       "2            8-Mar  \n",
       "3            Rasd1  \n",
       "4            Sgpp1  \n",
       "\n",
       "[5 rows x 935 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# python\n",
    "\n",
    "encodeA = pd.read_csv('/Users/Carol/Desktop/porra/ENCODE2016_UPDATE_PRONTO_FIM_transposed.csv',low_memory=False)\n",
    "#encodeB = pd.read_csv('/Users/Carol/Desktop/porra/ENCODE_TF_ChIP-seq_2015_trabalhado_overlap2016_transposed.csv',sep=',',header=True)\n",
    "\n",
    "encodeA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# list with the headers(name) of samples from list A\n",
    "goListA = list(encodeA)\n",
    "#goListA\n",
    "\n",
    "# list with the headers(name) of samples from list B\n",
    "goListB=list(encodeA)\n",
    "#goListB\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ListB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REST_H1hESC_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TAF1_HeLaS3_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REST_HepG2_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZBTB7A_K562_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SIX5_K562_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ListB\n",
       "0  REST_H1hESC_hg19\n",
       "1  TAF1_HeLaS3_hg19\n",
       "2   REST_HepG2_hg19\n",
       "3  ZBTB7A_K562_hg19\n",
       "4    SIX5_K562_hg19"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create dataframe from lists\n",
    "\n",
    "dfA= pd.DataFrame(goListA,columns=['ListA'])\n",
    "dfA.head()\n",
    "\n",
    "dfB = pd.DataFrame(goListB,columns=['ListB'])\n",
    "dfB.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>go.obj &lt;- newGeneOverlap(encodeA$REST_H1hESC_hg19</td>\n",
       "      <td>encodeB$REST_H1hESC_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>go.obj &lt;- newGeneOverlap(encodeA$TAF1_HeLaS3_hg19</td>\n",
       "      <td>encodeB$TAF1_HeLaS3_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>go.obj &lt;- newGeneOverlap(encodeA$REST_HepG2_hg19</td>\n",
       "      <td>encodeB$REST_HepG2_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>go.obj &lt;- newGeneOverlap(encodeA$ZBTB7A_K562_hg19</td>\n",
       "      <td>encodeB$ZBTB7A_K562_hg19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>go.obj &lt;- newGeneOverlap(encodeA$SIX5_K562_hg19</td>\n",
       "      <td>encodeB$SIX5_K562_hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   A                         B\n",
       "0  go.obj <- newGeneOverlap(encodeA$REST_H1hESC_hg19  encodeB$REST_H1hESC_hg19\n",
       "1  go.obj <- newGeneOverlap(encodeA$TAF1_HeLaS3_hg19  encodeB$TAF1_HeLaS3_hg19\n",
       "2   go.obj <- newGeneOverlap(encodeA$REST_HepG2_hg19   encodeB$REST_HepG2_hg19\n",
       "3  go.obj <- newGeneOverlap(encodeA$ZBTB7A_K562_hg19  encodeB$ZBTB7A_K562_hg19\n",
       "4    go.obj <- newGeneOverlap(encodeA$SIX5_K562_hg19    encodeB$SIX5_K562_hg19"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Acessing data # Defining overlap to be tested\n",
    "# Create a dataframe for tested where row has list A| listB\n",
    "\n",
    "# Append Columns to \"matrix\", \n",
    "go = []\n",
    "for row in dfA['ListA']:\n",
    "    a = 'go.obj <- newGeneOverlap(encodeA$%s' %(row)\n",
    "    go.append(a)\n",
    "go = pd.DataFrame(go)\n",
    "# go.head()\n",
    "\n",
    "gob = []\n",
    "for row in dfB['ListB']:\n",
    "    b = 'encodeB$%s'%(row) \n",
    "    gob.append(b)\n",
    "gob = pd.DataFrame(gob)\n",
    "# gob.head()\n",
    "\n",
    "## Concatenayed Columns\n",
    "gobject = pd.concat([go,gob],axis=1)\n",
    "gobject.columns = ['A', 'B']\n",
    "gobject.to_csv('gobject.csv',index=False) # save it to open in R in the other tab, since R stopped working here\n",
    "\n",
    "gobject.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# R code"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "### R code that used the previous dataframe to create the GeneOverlap command\n",
    "\n",
    "%%R\n",
    "\n",
    "for (i in gobject) {\n",
    "  b <- paste(sprintf('(%s,%s,genome.size=20805),',gobject$a,gobject$b))\n",
    "#   print(b)\n",
    "  write.table(b, \"enc15vs16freeOverlapMap2.csv\", quote=FALSE, row.names=F, col.names=FALSE, sep=\"\\t\")\n",
    "  }\n",
    "\n",
    "\n",
    "# Gives out \n",
    "#go.obj <- newGeneOverlap(encodeA$REST_H1hESC_hg19,encodeB$REST_H1hESC_hg19,,genome.size=20805)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "%%R\n",
    "\n",
    "# Load library\n",
    "library(GeneOverlap)\n",
    "\n",
    "# Read in R tables\n",
    "encodeA <- read.table('/Users/Carol/Desktop/porra/ENCODE2016_UPDATE_PRONTO_FIM_transposed.csv',sep=',', header=TRUE)\n",
    "encodeB <- read.table('/Users/Carol/Desktop/porra/ENCODE_TF_ChIP-seq_2015_trabalhado_overlap2016_transposed.csv',sep=',', header=TRUE)\n",
    "\n",
    "gobject <- read.table('/Users/Carol/Desktop/tfA/gobject.csv',sep=',', header=TRUE)\n",
    "\n",
    "# summary(gobject)\n",
    "#nrow(gobject)\n",
    "\n",
    "sapply(encodeA,length)\n",
    "sapply(encodeB,length)\n",
    "\n",
    "genome = 20805 # RefSeq Hg19 \n",
    "\n",
    "#go.obj <- newGeneOverlap(encodeA$REST_H1hESC_hg19,encodeB$REST_H1hESC_hg19,genome.size=genome)\n",
    "#print(go.obj)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "%%R\n",
    "# List of overlap comparisons. Ufahhhh...\n",
    "\n",
    "for (i in gobject) {\n",
    "  b <- paste(sprintf('(%s,%s,genome.size=20805)',gobject$a,gobject$b))\n",
    "#   print(b)\n",
    "  write.table(b, \"enc15vs16freeOverlapMap2.csv\", quote=FALSE, row.names=F, col.names=FALSE, sep=\"\\t\")\n",
    "  }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Overlap output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# Clean/ Organize/ Standardize output from gene overlap comparisons.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>category</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>listA</td>\n",
       "      <td>2000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>listB</td>\n",
       "      <td>1681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Intersection</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Union</td>\n",
       "      <td>3506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Overlapping p-value</td>\n",
       "      <td>0.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Odds ratio</td>\n",
       "      <td>1.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Jaccard</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>listA</td>\n",
       "      <td>2000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>listB</td>\n",
       "      <td>1174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Intersection</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              category value\n",
       "0               listA   2000\n",
       "1               listB   1681\n",
       "2        Intersection    175\n",
       "3               Union   3506\n",
       "4  Overlapping p-value  0.13\n",
       "5           Odds ratio   1.1\n",
       "6             Jaccard    0.0\n",
       "7               listA   2000\n",
       "8               listB   1174\n",
       "9        Intersection     98"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "# List of words to delete\n",
    "delete = ['size','#','\\n','Index',]\n",
    "\n",
    "# Data to keep \n",
    "words = ['listA', 'listB','Intersection','Overlapping p-value', 'Odds ratio','Union','Jaccard Index']\n",
    "\n",
    "\n",
    "rep = re.compile(r'|'.join(delete))\n",
    "keep = re.compile(r'|'.join(words))\n",
    "\n",
    "data = []\n",
    "with open('/Users/Carol/Desktop/tfA/ENCODEE15E15.csv') as f1:\n",
    "    for line in f1:\n",
    "        if keep.search(line):\n",
    "            # replace all unwanted substrings\n",
    "            line = rep.sub(\"\", line.lstrip())     \n",
    "            line=line.split('=')\n",
    "#             print(line)\n",
    "            data.append(line)\n",
    "\n",
    "#print df\n",
    "# type(data)\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "df.columns = ['category', 'value']\n",
    "df['value'] = df.value.str.split(',').str.get(0)  # split value column at the , and que left side of string\n",
    "# df.head(100)\n",
    "\n",
    "\n",
    "# grouped = df.groupby('category')\n",
    "# grouped.head()\n",
    "\n",
    "df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "labels=df['category']\n",
    "labels = labels.drop_duplicates()\n",
    "labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "values=df\n",
    "values\n",
    "# df.loc[df['column_name'] == some_value]\n",
    "# encode_metadata = encode_metadata[~encode_metadata.Biosample_treatments.isin(value_list_treatment)]\n",
    "\n",
    "# listA = df[(df[1] == 'listA')]\n",
    "# listB \n",
    "# Intersection \n",
    "# Union \n",
    "# Overlapping p-value\n",
    "# Odds ratio\n",
    "# Jaccard Index\n",
    "\n",
    "df\n",
    "listA = df[df['category'].str.contains(\"listA\")]\n",
    "listB = df[df['category'].str.contains(\"listB\")]\n",
    "intersection = df[df['category'].str.contains(\"Intersection\")]\n",
    "union = df[df['category'].str.contains(\"Union\")]\n",
    "pval= df[df['category'].str.contains(\"Overlapping p-value\")]\n",
    "odds= df[df['category'].str.contains(\"Odds ratio\")]\n",
    "jaccard= df[df['category'].str.contains(\"Jaccard Index\")]\n",
    "\n",
    "\n",
    "# Dropped category column and just kept the values column\n",
    "# listA = listA.drop('category', axis=1)\n",
    "listA = listA.T\n",
    "\n",
    "\n",
    "# listB = listB.drop('category', axis=1)\n",
    "listB = listB.T\n",
    "\n",
    "# intersection = intersection.drop('category', axis=1)\n",
    "Intersection = intersection.T\n",
    "\n",
    "# union = union.drop('category', axis=1)\n",
    "Union = union.T\n",
    "\n",
    "# pval = pval.drop('category', axis=1)\n",
    "pval = pval.T\n",
    "\n",
    "# odds = odds.drop('category', axis=1)\n",
    "odds = odds.T\n",
    "\n",
    "# jaccard = jaccard.drop('category', axis=1)\n",
    "# Jaccard Index = jaccard.T\n",
    "\n",
    "# dff = listA.append(listB)\n",
    "listA = listA.drop('category')\n",
    "listA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "grouped = df.as_matrix(df, ncol=7, byrow=True)\n",
    "grouped = df.groupby('category')\n",
    "grouped.head()\n",
    "\n",
    "\n",
    "### DICT desses dados#####\n",
    "###nao usei \n",
    "#Created dictionary with the data\n",
    "# data = pd.DataFrame({'category': ['listA', 'listB', 'Intersection', 'Union','Overlapping','Odds','Jaccard']}) \n",
    "# #*4, 'Ob1' : np.random.rand(16), 'Ob2' : np.random.rand(16)})\n",
    "# data\n",
    "\n",
    "# #create unique list of names\n",
    "# uniqueCat = df.category.unique()\n",
    "\n",
    "# # #create a data frame dictionary to store your data frames\n",
    "# # DataFrameDict = { : pd.DataFrame for elem in uniqueCat}\n",
    "\n",
    "# for key in DataFrameDict.keys():\n",
    "#     DataFrameDict[key] = df[:][df.category == key]\n",
    "    \n",
    "# # DataFrameDict"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
